src/UI/PersonNewsFeedBar.jsx

```diff
@@ -1,7 +1,16 @@
-import { useState } from "react";
+import {
+  Button,
+  Checkbox,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+} from "@mui/material";
+import { useEffect, useRef, useState } from "react";
 import { useNavigate } from "react-router-dom";
 
-const PreferencesForm = () => {
+const PreferencesForm = ({ isOpen, toggleModal }) => {
```

Comment:
Few suggestions here:|
 1. Make component name descriptive and reflect it's purpose, I'd call this `NewsSourcesModal`
 2. If you were to include the search input in this modal as well, then I'd name it `SearchModal` (bit more generic, but still descriptive)
 3. The file and component should always have the same name, so example component: `SearchBar`, file: `SearchBar.jsx`